### PR TITLE
add filter params to raw()

### PR DESCRIPTION
**Pull request by tgib23**
This PR adds filter params to raw() method to support the same params as list() does.

Reference : #820

### CODE REVIEW

File: tests/tests_integration/test_api/test_raw.py [hunk 1]
```diff
@@ -45,12 +45,18 @@ def test_list_rows(self):
...
-        assert 10 == len(rows)
+        rows_list = COGNITE_CLIENT.raw.rows.list(
+            db_name="test__database1", table_name="test__table_1", limit=10, columns=["c1", "c2"]
+        )
+        assert 10 == len([x for x in rows])
+        assert 10 == len(rows_list)
         for row in rows:
             assert {"c1", "c2"} == set(row.columns.keys())
+        for row in rows_list:
+            assert {"c1", "c2"} == set(row.columns.keys())
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_raw.py:59]: Better to split it into a separate test: `test_iter_rows_cols`

File: tests/tests_unit/test_api/test_raw.py [hunk 2]
```diff
@@ -268,6 +268,26 @@ def test_delete_fail(self, rsps):
             RAW_API.rows.delete("db1", "table1", "key1")
         assert e.value.failed == ["key1"]
 
+    def test_row(self, mock_raw_row_response):
```
Comment (erlendvollset) [inline tests/tests_unit/test_api/test_raw.py:271]:
```suggestion
    def test_iter(self, mock_raw_row_response):
```

File: tests/tests_unit/test_api/test_raw.py [hunk 3]
```diff
@@ -268,6 +268,26 @@ def test_delete_fail(self, rsps):
...
+        res_generator = RAW_API.rows(db_name="db1", table_name="table1")
+        row = next(res_generator)
+        assert Row(key="row1", columns={"c1": 1, "c2": "2"}) == row
+        assert "columns=" not in mock_raw_row_response.calls[0].request.path_url
+
+    def test_rows_cols(self, mock_raw_row_response):
```
Comment (erlendvollset) [inline tests/tests_unit/test_api/test_raw.py:277]:
```suggestion
    def test_iter_cols(self, mock_raw_row_response):
```

File: tests/tests_unit/test_api/test_raw.py [hunk 4]
```diff
@@ -268,6 +268,26 @@ def test_delete_fail(self, rsps):
...
+        res_generator = RAW_API.rows(db_name="db1", table_name="table1", columns=["a", 1])
+        next(res_generator)
+        assert "columns=a%2C1" in mock_raw_row_response.calls[0].request.path_url
+
+    def test_rows_cols_empty(self, mock_raw_row_response):
```
Comment (erlendvollset) [inline tests/tests_unit/test_api/test_raw.py:282]:
```suggestion
    def test_iter_cols_empty(self, mock_raw_row_response):
```

File: tests/tests_unit/test_api/test_raw.py [hunk 5]
```diff
@@ -268,6 +268,26 @@ def test_delete_fail(self, rsps):
...
+        res_generator = RAW_API.rows(db_name="db1", table_name="table1", columns=[])
+        next(res_generator)
+        assert "columns=%2C&" in mock_raw_row_response.calls[0].request.path_url + "&"
+
+    def test_rows_cols_str_not_supported(self, mock_raw_row_response):
```
Comment (erlendvollset) [inline tests/tests_unit/test_api/test_raw.py:287]:
```suggestion
    def test_iter_cols_str_not_supported(self, mock_raw_row_response):
```

File: cognite/client/_api/raw.py [hunk 6]
```diff
@@ -283,12 +290,28 @@ def __call__(
...
+            max_last_updated_time (int): Rows must have been last updated before this time. ms since epoch.
+            columns (List[str]): List of column keys. Set to `None` for retrieving all, use [] to retrieve only row keys.
         """
+        if columns is not None:
+            if not isinstance(columns, List):
+                raise ValueError("Expected a list for argument columns")
+            if len(columns) == 0:
+                columns = ","
+            else:
+                columns = ",".join([str(x) for x in columns])
```
Comment (erlendvollset) [inline cognite/client/_api/raw.py:303]: Code duplicated in `list()` method, extract into a protected method.

**Review by erlendvollset — CHANGES_REQUESTED** [review]
Nice! Thanks for doing this!

**Comment by tgib23** [comment]
@erlendvollset thanks for your review. I think I've fixed the whole you pointed out.

File: cognite/client/_api/raw.py [hunk 7]
```diff
@@ -470,3 +483,12 @@ def list(
                 "columns": columns,
             },
         )
+
+    def _make_columns_param(self, columns: List[str]) -> str:
+        if not isinstance(columns, List):
```
Comment (erlendvollset) [inline cognite/client/_api/raw.py:488]:
```suggestion
    def _make_columns_param(self, columns: Optional[List[str]]) -> Optional[str]:
        if columns is None:
            return None
        if not isinstance(columns, List):
```

Comment (erlendvollset) [reply cognite/client/_api/raw.py:488]: And then remove the `if columns is not None` check from `list()` and `__iter__()`

**Review by erlendvollset — COMMENTED** [review]
Just some nitpick. Please update the changelog aswell (and bump version number if you want to trigger a new release)

**Comment by tgib23** [comment]
@erlendvollset I've pushed the changes. Please let me know if there is anything else to change.

**Review by erlendvollset — APPROVED** [review]
LGTM. You need to update the version in cognite/client/_version.py in order for a new release to be made.

File: CHANGELOG.md [hunk 8]
```diff
@@ -14,6 +14,11 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [2.19.1] - 2021-05-27
```
Comment (erlendvollset) [inline CHANGELOG.md:17]:
```suggestion
## [2.19.1] - 2021-06-04
```

**Comment by erlendvollset** [comment]
> LGTM. You need to update the version in cognite/client/_version.py in order for a new release to be made.

I updated it now.

**Comment by tgib23** [comment]
@erlendvollset thanx for your help!

**Pull request merged**